src/architecture/storage.md

```diff
@@ -1,262 +1,158 @@
-  storing big and complex elements.
-- The `KeySetIndex` introduces a lexicographical order over stored
-  elements, while the `ValueSetIndex` orders elements arbitrarily due to hash
-  function properties.
-
-### Entry
+This page provides an overview of how Exonum works with the persistent data.
 
-`Entry` represents an index that contains only one element.
-
-The following operations are implemented:
-
-- Get, set and remove value
-- Check if the value is present
-
-### Merkelized Indices
-
-Merkelized indices represent a list and a map with additional
-features. Such indices can create the proofs of existence or absence for
-stored data items.
+Exonum uses MerkleDB as a storage framework. MerkleDB is an object database. It
+represents a high-level wrapper over the key-value store.
 
-When a light client requests data from an Exonum full node, the proof can be
-built and sent along with the actual data. Having block headers
-and this proof, the client may check that received data was really
-authorized by the validators without having to replicate the entire blockchain
-contents.
-
-#### ProofListIndex
-
-[`ProofListIndex`][proof-list-index] implements a Merkle
-tree, which is a Merkelized version of an
-array list. It implements the same methods as `ListIndex`, and adds an
-additional feature: based on Merkle trees, `ProofListIndex` allows efficiently
-creating compact proofs of existence for the list items.
-The following additional procedures are implemented:
+Currently [RocksDB][rocks-db] is used for low-level key-value storage engine in
+Exonum.
 
-- Get the height of the Merkle tree. As the tree is balanced (though may be not
-  full), its height is close to `log2` of the list length
-- Get the value of the tree root (i.e., the hash of the entire Merkle tree)
-- Build a proof of existence for an item at a specific position
-- Build a proof of existence for items at a specific contiguous index range
+The objects of MerkleDB are convenient abstractions for work with
+blockchain-specific data. For example, such abstraction can be a list that
+provides cryptographic proofs for its stored items.
 
-!!! note
-    `ProofListIndex` is *append-only*; it does not allow deleting list items.
-    The only way to delete an item from a `ProofListIndex` is clearing it.
+Currently, the objects in MerkleDB fall into two types:
 
-!!! summary "Implementation Details"
-    As with `ListIndex`, list items are stored with 8-byte keys. However,
-    `ProofListIndex` also persists all intermediate nodes of the Merkle tree
-    built on top of the list, in order to quickly build proofs and recalculate
-    the Merkle tree after operations on the list.
+- **blobs**, which represent sequences of bytes, and
+- **root objects**, that do not have parents. These objects have UTF-8
+  identifiers, for example, "block", "state". Root objects can contain blob
+  items inside them.
 
-#### ProofMapIndex
+The basic root objects of the framework are:
```

Comment:
Is it really needed to use 'root object' at this point of the db development? Will non-root object be of different types (**not** lists, maps, …)? If there is nothing else, shall a user care if they are 'root'? 

Do I also have multiple roots — a forest of these objects?

Reply:
Here we need to explicitly distinguish between blobs and other objects. Besides, in the future, if I understand correctly, the term "root object" will also distinguish the object that has no parents from other objects.

There can be multiple root objects within one service.